db/functions/hints.py

```diff
@@ -101,6 +101,9 @@ def _is_hint_applicable_to_types(hint):
 array = _make_hint("array")
 
 
+numeric = _make_hint('numeric')
```

Comment:
Can you rename `numeric` to `integer`? A decimal would not be appropriate when specifying the index of a dimension, but it would still be numerical.